### PR TITLE
Token handling and general refactoring/optimization

### LINKED ISSUE #1: Implement token storage and consolidate storage `kind`
Using the default `local-storage` plugin has the following behavior:
1. JWT signing secret is stored to `<storage_path>/.verdaccio-db.json` under a top level key of `secret` in the JSON file.
2. The list of published packages is stored to `<storage_path>/.verdaccio-db.json` under a top level key of `list` in the JSON file. This list is a simple array of published package names (only names, no version info). When the Verdaccio UI loads and displays the list of published packages it uses this list when building the results.
3. JWT token metadata is stored to `<storage_path>/.token-db.json`. This file contains metata on each issued JWT token (NOT the full JWT token) and is used to provide responses to the `npm token` command.

The existing `verdaccio-google-storage` plugin has the following behavior:
1. JWT signing secret is stored to GCP Datastore under a `kind` named "*Secret*". The secret is the only entity stored under this `kind` and has a single "*secret*" property where the value of the JWT signing secret value resides.
2. The list of published packages is stored to GCP Datastore under a `kind` named "*VerdaccioDataStore*" (this is the default name but is configurable). Each published package name is stored as a separate entity with a single property named "*name*" that holds the package name. When the Verdaccio UI loads and displays the list of published packages it does a query against this `kind` with no filters as part of building the list.
3. JWT token metadata related storage is NOT implemented at all.

Because the current `verdaccio-google-storage` plugin lacks implementation of the necessary token actions to store/retrieve user token metadata,  the `npm create token` command fails. Additional details:
* In order to generate tokens and store token metadata the following functions must be implemented in a storage plugin: https://github.com/verdaccio/monorepo/blob/v10.0.6/core/types/index.d.ts#L377
* The `saveToken`, `deleteToken`, and `readTokens` functions are implemented in the default `local-storage` plugin: https://github.com/verdaccio/monorepo/blob/v10.0.6/plugins/local-storage/src/token.ts
* These functions explicitly reject in the `verdaccio-google-cloud` storage plugin: https://github.com/verdaccio/monorepo/blob/v10.0.6/plugins/google-cloud/src/data-storage.ts#L70

## Proposal
* Consolidate all data storage down to a single GCP Datastore `kind`, adding additional metadata to each entity stored so records can be filtered based on the type of record being retrieved.
* Implement the `saveToken`, `deleteToken`, and `readTokens` functions. The token metadata records would be stored under the same `kind` used for storing package names. An additional "*type*" property will be added to the entity with a static value of `token` so when retrieving/listing tokens a filter of `type=token` will be applied.
* Refactor the current storage of package names to add an additional "*type*" property to the entity with a static value of `package` so when listing package names a filter of `type=package` will be applied.
* One of two possible changes to the implementation of the JWT signing secret:
  1. **Preferred** - Use GCP Secret Manager for storing/retrieving the JWT signing secret. This would be more secure then using GCP Datastore and would allow for better IAM management of the JWT signing secret. Right now anyone who has read access to GCP Datastore can view and potentially modify this secret, breaking customer tokens.
  2.  Refactor the current storage of JWT secret to add an additional "*type*" property to the entity with a static value of `secret` so when retrieving the secret a filter of `type=secret` will be applied.

### LINKED ISSUE #9: Update License
- Keep the MIT license
- Keep the existing Copyright
- Add `Copyright (c) 2021 XLTS.dev` below the existing Copyright

### PR DESCRIPTION
**Pull request by edclement**
Closes #1, Closes #9

Follow-up Issues have been opened for:
- Unit tests - https://github.com/xlts-dev/verdaccio-google-cloud-storage/issues/3
- Search - https://github.com/xlts-dev/verdaccio-google-cloud-storage/issues/4
- data-storage.ts clean-up - https://github.com/xlts-dev/verdaccio-google-cloud-storage/issues/5
- IAC - https://github.com/xlts-dev/registry/issues/30

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "verdaccio-google-cloud",
-  "version": "10.0.1",
+  "name": "verdaccio-google-cloud-storage",
```
Comment (Splaktar) [inline package.json:2]:
```suggestion
  "name": "@xlts.dev/verdaccio-google-cloud-storage",
```

File: package.json [hunk 2]
```diff
@@ -10,49 +10,60 @@
     "datastore"
   ],
   "author": "Juan Picado <juanpicado19@gmail.com>",
+  "contributors": [{
+    "name": "Ed Clement",
+    "email": "ed.clement@devintent.com",
```
Comment (Splaktar) [inline package.json:15]:
```suggestion
    "email": "ed@xlts.dev",
```

File: package.json [hunk 3]
```diff
@@ -10,49 +10,60 @@
...
+    "@types/node": "14.14.31",
+    "@verdaccio/babel-preset": "10.0.0",
+    "@verdaccio/eslint-config": "10.0.0",
+    "@verdaccio/types": "10.0.0",
+    "babel-eslint": "10.1.0",
+    "cross-env": "7.0.3",
+    "eslint": "6.8.0",
+    "jest": "26.0.1",
+    "lint-staged": "9.5.0",
+    "prettier": "1.19.1",
```
Comment (Splaktar) [inline package.json:53]: This is a pretty old version. Can you please open an issue to update prettier to a newer release? It's better to do it in a separate PR to keep this PR lean.

Comment (edclement) [reply package.json:53]: opened #7
